content/ADR-153.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: adr
+adr: 153
+title: Semantics of entity parenting with ECS-compatible semantics
+date: 2022-12-14
+status: Living
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - menduz
+---
+
+## Abstract
+
+This document describes the semantics of changing parenting of entities and what happens when an entity in the middle of a branch of the tree is removed. The problem is complex because in an ECS approach, entities are stored in a flat structure, and trees are a synthetic construct for positioning purposes only. The decision is that when an entity in the middle of a branch is removed, all its children will be re-parented to the root entity.
+
+#### Assumptions
+
+- When a `Transform` component doesn't exist in an entity, it is assumed that the entity has a `scale={1,1,1}`
+- Thus, it is safe to assume that a missing `Transform` component has the same rendering semantics than an entity with a `Transform.Identity`
+- And since entities are defined by the set of components and don't exist by themselves. It is safe to assume that all entities have an `Transform.Identity` by default.
```

Comment:
```suggestion
- And since entities are defined by the set of components and don't exist by themselves, it is safe to assume that all entities have a `Transform.Identity` by default.
```